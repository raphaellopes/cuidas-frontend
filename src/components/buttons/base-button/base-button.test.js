// vendors
import React from 'react';
import renderer from 'react-test-renderer';

// locals
import BaseButton from './index';

const component = props => (renderer.create(
  <BaseButton {...props} />,
));

describe('<BaseButton />', () => {
  test('Should render the component', () => {
    const tree = component({ children: 'My children' }).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should change the color when prop is set', () => {
    const props = { children: 'My children', color: 'primary' };
    const tree = component(props).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should change the backgroundColor when prop is set', () => {
    const props = { children: 'My children', backgroundColor: 'primary' };
    const tree = component(props).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
