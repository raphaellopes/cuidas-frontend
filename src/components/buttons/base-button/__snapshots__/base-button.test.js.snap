// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseButton /> Should change the backgroundColor when prop is set 1`] = `
<button
  className="sc-bdVaJa dOxXJD"
  type="button"
>
  My children
</button>
`;

exports[`<BaseButton /> Should change the color when prop is set 1`] = `
<button
  className="sc-bdVaJa kebtMc"
  color="primary"
  type="button"
>
  My children
</button>
`;

exports[`<BaseButton /> Should render the component 1`] = `
<button
  className="sc-bdVaJa debtoM"
  type="button"
>
  My children
</button>
`;
